packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -146,8 +146,9 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   /// [LocalizationsDelegate] implementations typically call the static [load]
   /// function, rather than constructing this class directly.
   DefaultMaterialLocalizations(this.locale)
-      : assert(locale != null),
-        this._localeName = _computeLocaleName(locale) {
+      : this._localeName = _computeLocaleName(locale) {
+    // TODO(tvolkert): Move this assert back to the initializer.
```

Comment:
since we have a body here it really doesn't matter where the assert is, so you can remove the TODO.